src/Streamly/Prelude.hs

```diff
@@ -166,6 +168,16 @@ foldl step begin done m = go begin (toStream m)
             yield a (Just x) = go (step acc a) x
          in (runStream m1) Nothing stop yield
 
+-- | Scan left. A strict left fold which accumulates the result of its reduction steps inside a stream, from left.
+scanl :: (Streaming t, Monad m) => (b -> a -> b) -> b -> t m a -> t m b
+scanl f seed m = cons seed $ fromStream $ go (toStream m) seed
+    where
+    go m1 !s = Stream $ \_ stp yld ->
```

Comment:
call it `acc` (accumulator) instead of `s` to be consistent with `foldl` implementation.